### PR TITLE
#12376: Support for non-32 Height in Width Sharded Conv2d

### PR DESCRIPTION
**Pull request by sankarmanoj-tt**
### Ticket
#12376 

### Problem description
Support for non-32 Height in Width Sharded Conv2d

### Checklist
- [ ] Post commit CI [passes](https://github.com/tenstorrent/tt-metal/actions/runs/10809220609)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

File: ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.cpp [hunk 1]
```diff
@@ -204,13 +204,13 @@ OptimizedConvParallelizationConfig determine_conv_op_parallel_config_from_conv_o
     TT_ASSERT(conv_output_mem_config.shard_spec.has_value());
     const auto& shard_spec = conv_output_mem_config.shard_spec.value();
     const auto& shard_shape = shard_spec.shape;
-    TT_ASSERT(shard_shape[0] % 32 == 0);
+    // TT_ASSERT(shard_shape[0] % 32 == 0);
```
Comment (mywoodstock) [inline ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.cpp:207]: delete

Comment (mywoodstock) [reply ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.cpp:207]: does this affect height and block sharding variants?

File: tests/ttnn/unit_tests/operations/test_new_conv2d.py [hunk 2]
```diff
@@ -317,18 +317,21 @@ def run_conv_with_split(
     assert_with_pcc(torch_output_tensor, torch_out_golden_tensor, pcc=pcc)
 
 
+@skip_for_grayskull()
```
Comment (bbradelTT) [inline tests/ttnn/unit_tests/operations/test_new_conv2d.py:320]: Do other tests cover these test scenarios on grayskull?
If not, why would these test scenarios note be relevant for grayskull?

Comment (sankarmanoj-tt) [reply tests/ttnn/unit_tests/operations/test_new_conv2d.py:320]: We are not planning on supporting these new features on gray skull.

File: ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.cpp (hunk 1 above)
Comment (sankarmanoj-tt) [reply ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.cpp]: Fixed.

File: tests/ttnn/unit_tests/operations/test_new_conv2d.py (hunk 2 above)
Comment (bbradelTT) [reply tests/ttnn/unit_tests/operations/test_new_conv2d.py:320]: Are the cases in the already existing test (before this PR) supported by grayskull?

Comment (sankarmanoj-tt) [reply tests/ttnn/unit_tests/operations/test_new_conv2d.py:320]: Most of them were skipped, due to an OOM error.  The width sharded Conv2d which is tested by this code was implemented to support a wormhole only model.

**Review by bbradelTT — APPROVED** [review]

**Review by mywoodstock — APPROVED** [review]

**Pull request merged**